### PR TITLE
sam9x60 sip types added to soc description.

### PR DESCRIPTION
**Pull request by ksdd**
Hi,

this patch adds the SAM9X60 SIP variants to the soc description list.

### CODE REVIEW

File: drivers/soc/atmel/soc.c [hunk 1]
```diff
@@ -237,7 +243,8 @@ struct soc_device * __init at91_soc_init(const struct at91_soc *socs)
 	}
 
 	if (!soc->name) {
-		pr_warn("Could not find matching SoC description\n");
+		pr_warn("Could not find matching SoC description"
+			" (cidr %08x, exid %08x)", cidr, exid);
 		return NULL;
```
Comment (alexandrebelloni) [inline drivers/soc/atmel/soc.c:248]: This is an unrelated change

**Review by alexandrebelloni — COMMENTED** [review]
Seems good to me, maybe you should also submit that upstream.

File: drivers/soc/atmel/soc.c (hunk 1 above)
Comment (ksdd) [reply drivers/soc/atmel/soc.c:248]: Should I revert the change of the warning and probably create a different pull request? It is definitely useful, when running on yet unsupported soc's.

Comment (alexandrebelloni) [reply drivers/soc/atmel/soc.c:248]: Yes, two separate patches would be the best. However, This will increase the kernel size for a very unusual use case. Most of the time, the soc ids will be present. Maybe use pr_debug ?

BTW, are you actually using this driver for something useful?

Comment (ksdd) [reply drivers/soc/atmel/soc.c]: I reverted the change of the print and decided not to create a different pull request for it. As You pointed out, it should be of rare use. To be honest, I don't know, what the soc registration is good for. I just stumbled over the ""Could not find matching SoC description" warning in the dmesg and fixed it.

You proposed to submit this change upstream. I guess You mean submitting it to the mainline kernel. I requested the pull to linux-at91, because this is my direct upstream. And I thought, that the linux-at91 maintainers will submit relevant changes further up.

Comment (noglitch) [reply drivers/soc/atmel/soc.c]: Yes, we can do it.

**Comment by noglitch** [comment]
BTW @ksdd can you add your Signed-off-by: tag to the patch so that I can add it.
You would also need to tell us if you want this S-o-B tag to appear in kernel.org upstream.
Best regards,
  Nicolas

File: drivers/soc/atmel/soc.c [hunk 2]
```diff
@@ -69,6 +69,12 @@ static const struct at91_soc __initconst socs[] = {
 #endif
 #ifdef CONFIG_SOC_SAM9X60
 	AT91_SOC(SAM9X60_CIDR_MATCH, SAM9X60_EXID_MATCH, "sam9x60", "sam9x60"),
+	AT91_SOC(SAM9X60_CIDR_MATCH, SAM9X60D5M_EXID_MATCH,
+		 "sam9x60d6m", "sam9x60"),
```
Comment (noglitch) [inline drivers/soc/atmel/soc.c:73]: Take example on sama5d2 SiP lines to have a coherent display between the different SiP products.

**Review by noglitch — CHANGES_REQUESTED** [review]

File: drivers/soc/atmel/soc.c (hunk 2 above)
Comment (ksdd) [reply drivers/soc/atmel/soc.c]: Hi @noglitch, I changed the descriptions as You suggested. I hope it is okay now.

**Comment by ksdd** [comment]
Hi @noglitch, I changed the descriptions as You suggested. I hope, it is okay now.

**Comment by ksdd** [comment]
Regarding "Signed-off-by", I am not sure, how to do that for already pushed commits. I could re-base back to branch start and re-commit the changes with S-o-B. But that requires a force push and changes commit history, which should usually be avoided. Is S-o-B really needed?

**Comment by noglitch** [comment]
Well, simply add :

File: drivers/soc/atmel/soc.c [hunk 3]
```diff
@@ -70,11 +70,11 @@ static const struct at91_soc __initconst socs[] = {
 #ifdef CONFIG_SOC_SAM9X60
 	AT91_SOC(SAM9X60_CIDR_MATCH, SAM9X60_EXID_MATCH, "sam9x60", "sam9x60"),
 	AT91_SOC(SAM9X60_CIDR_MATCH, SAM9X60D5M_EXID_MATCH,
-		 "sam9x60d6m", "sam9x60"),
+		 "sam9x60d5m 512MiB DDR2 SiP", "sam9x60"),
 	AT91_SOC(SAM9X60_CIDR_MATCH, SAM9X60D1G_EXID_MATCH,
-		 "sam9x60d1g", "sam9x60"),
+		 "sam9x60d1g 1GiB DDR2 SiP", "sam9x60"),
```
Comment (noglitch) [inline drivers/soc/atmel/soc.c:75]: Well, no, this is not 1GiB, it's 128MiB: please correct all them.

File: drivers/soc/atmel/soc.c [hunk 4]
```diff
@@ -70,11 +70,11 @@ static const struct at91_soc __initconst socs[] = {
 #ifdef CONFIG_SOC_SAM9X60
 	AT91_SOC(SAM9X60_CIDR_MATCH, SAM9X60_EXID_MATCH, "sam9x60", "sam9x60"),
 	AT91_SOC(SAM9X60_CIDR_MATCH, SAM9X60D5M_EXID_MATCH,
```
Comment (noglitch) [inline drivers/soc/atmel/soc.c:72]: I'd like that the #defines are like the other ones: SAM9X60_D5M_EXID_MATCH

**Review by noglitch — CHANGES_REQUESTED** [review]

File: drivers/soc/atmel/soc.c (hunk 3 above)
Comment (ksdd) [reply drivers/soc/atmel/soc.c:75]: Bits and bytes - always difficult. I will fix it.

**Comment by alexandrebelloni** [comment]
You should definitively rebase and squash all those commits and also add your SoB. Rebasing is ok because you don't have any downstream.

**Comment by ksdd** [comment]
@alexandrebelloni : Okay, I will rebase and squash, once @noglitch is happy with the changes.

File: drivers/soc/atmel/soc.c (hunk 4 above)
Comment (ksdd) [reply drivers/soc/atmel/soc.c]: @noglitch : EXID_MATCH definitions changed with commit feff97a0e7052ec8d397d953d85424a15f25fb87.

**Comment by noglitch** [comment]
Patch applied in linux-5.4-at91 branch and master. Thanks a lot Kai for patiently having re-spin this patch.
https://github.com/linux4sam/linux-at91/commit/e60a4f2ce49bd1d264d97b6b49fe879fd3644fe4
Best regards,
  Nicolas
